docs/source/notes/amp_examples.rst

```diff
@@ -0,0 +1,210 @@
+.. _amp-examples:
+
+Automatic Mixed Precision Examples
+==================================
+
+When following the examples below, you don't need to call ``.half()`` on your model(s) or data.
+In fact, you shouldn't.  Model weights should remain FP32.
+
+You also don't need to retune any hyperparameters.
+
+.. contents:: :local:
+
+.. _autocasting-examples:
+
+Autocasting Examples
+^^^^^^^^^^^^^^^^^^^^
+
+Under construction...
+
+.. _gradient-scaling-examples:
+
+Gradient Scaling Examples
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The code snippets below demonstrate recommended use of :class:`torch.cuda.amp.AmpScaler`.
+
+.. currentmodule:: torch.cuda.amp
+
+Typical use (1 loss, 1 optimizer)
+---------------------------------
+
+::
+
+    scaler = AmpScaler()
+    ...
+    for input, target in data:
+        optimizer.zero_grad()
+        output = model(input)
+        loss = loss_fn(output, target)
+        scaler.scale(loss).backward()
+        scaler.step(optimizer)
+        scaler.update()
+
+
+Gradient clipping
+-----------------
+
+Gradient clipping requires awareness that the gradients resulting from ``scaler.scale(loss).backward()`` are scaled.
```

Comment:
Introduce gradient clipping briefly. Not just gradient clipping is relevant here. This section might be better as "Working with Unscaled Gradients." Then you explain that using the AMPScaler, as mentioned above (mention this above), scales gradients to fit within torch.half's representable range. To manipulate the unscaled gradients do the following...